contracts/test/PlonkVerifier.t.sol

```diff
@@ -28,7 +28,7 @@ contract PlonkVerifierCommonTest is Test {
     /// @dev Sanitize a single value to be valid scalar field Bn254::Fr.
     function sanitizeScalarField(uint256 a) public view returns (uint256) {
         a = bound(a, 0, BN254.R_MOD - 1);
-        BN254.validateScalarField(a);
+        BN254.validateScalarField(BN254.ScalarField.wrap(a));
         return a;
```

Comment:
should we just retern a `ScalarField` instead?

Reply:
I tried but it does not seem to help. By doing such modification we kind of end up having to change the signature of `PlonkVerifier.verify`  from  
* `PlonkVerifier.verify(IPlonkVerifier.VerifyingKey memory verifyingKey,   uint256[] memory publicInput,        IPlonkVerifier.PlonkProof memory proof,   bytes memory extraTranscriptInitMsg)`
to    
*   `PlonkVerifier.verify(IPlonkVerifier.VerifyingKey memory verifyingKey,   BN254.ScalarField[] memory publicInput,   IPlonkVerifier.PlonkProof memory proof,   bytes memory extraTranscriptInitMsg)`.
        
But I think that the original signature is fine as indeed the input could be arbitrary/malformed.

Reply:
what? the public inputs are supposed to be ScalarField right? why could they be anything else?

(altho I agree that just marking it as ScalarField doesn't automatically give you value integrity, as in someone can still wrap a value > p as inputs)

Reply:
See #859